src/components/modals/ModalUserEdit.tsx

```diff
@@ -0,0 +1,363 @@
+import { Box, Button, Grid, Modal, SimpleGrid, Space, Text, TextInput, Title } from "@mantine/core";
+import React, { useEffect, useRef, useState } from "react";
+import { useTranslation } from "react-i18next";
+import SortableTree from "react-sortable-tree";
+import FileExplorerTheme from "react-sortable-tree-theme-file-explorer";
+import { Mail, User } from "tabler-icons-react";
+
+import { signup } from "../../actions/authActions";
+import { fetchDirectoryTree, manageUpdateUser, updateUserAndScan } from "../../actions/utilActions";
+import { useAppDispatch, useAppSelector } from "../../store/store";
+import { PasswordEntry } from "../settings/PasswordEntry";
+
+type Props = {
+  isOpen: boolean;
+  updateAndScan?: boolean;
+  userToEdit: any;
+  selectedNodeId?: string;
+  onRequestClose: () => void;
+  userList: any;
+  createNew: boolean;
+  firstTimeSetup?: boolean;
+};
+
+export function ModalUserEdit(props: Props) {
+  const { isOpen, updateAndScan, selectedNodeId, onRequestClose, userList, createNew, firstTimeSetup } = props;
+  var userToEdit = props.userToEdit;
+  const [newScanDirectory, setNewScanDirectory] = useState("");
+  const [treeData, setTreeData] = useState([]);
+  const [userName, setUserName] = useState("");
+  const [userNameError, setUserNameError] = useState("");
+  const [userEmail, setUserEmail] = useState("");
+  const [userEmailError, setUserEmailError] = useState("");
+  const [userFirst, setUserFirst] = useState("");
+  const [userLast, setUserLast] = useState("");
+  const [userPassword, setUserPassword] = useState("");
+  const [newPasswordIsValid, setNewPasswordIsValid] = useState(true);
+
+  const [isAdmin, setIsAdmin] = useState(false);
+  const [scanDirectoryPlaceholder, setScanDirectoryPlaceholder] = useState("");
+  const dispatch = useAppDispatch();
+  const auth = useAppSelector(state => state.auth);
+  const { directoryTree } = useAppSelector(state => state.util);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const { t } = useTranslation();
+  const [pathDoesNotExist, setPathDoesNotExist] = useState(false);
+  const [pathError, setPathError] = useState("");
+  const [closing, setClosing] = useState(false);
+
+  useEffect(() => {
+    if (auth.access && auth.access.is_admin) {
+      dispatch(fetchDirectoryTree(""));
+    }
+  }, [auth.access, dispatch]);
+
+  useEffect(() => {
+    if (treeData.length === 0) {
+      setTreeData(directoryTree);
+    } else {
+      const newData = replacePath(treeData, directoryTree[0]);
+
+      // @ts-ignore
```

Comment:
why the `@ts-ignore` here?

Reply:
This is code from the legacy component. So this would be an optional fix from my point of view.